### DISCUSSION TITLE
How to transform data to use pseudoRes

### DISCUSSION THREAD

**Discussion by AmandaPS22** [bmcclintock/momentuHMM#96] [Q&A] [answered]
Hi guys, I'm processing data for a kernel bivariate probability distribution (kde) and I'm trying to use the pseudoRes function but the following error:

_Error in pseudoRes(df):
   'm' must be a momentuHMM, HMMfits, miHMM, or miSum object (as output by fitHMM, MIfitHMM, or MIpool)_

df is a dataframe.
How do I transform my kde results to use the pseudoRes function?

Thanks

**Comment by bmcclintock** [comment, marked as answer]
`pseudoRes` is a function for calculating pseudo-residuals for a hidden Markov model (HMM) fitted using the `momentuHMM` functions `fitHMM` or `MIfitHMM`. You would need to formulate your model as a HMM and fit it using `fitHMM` (or `MIfitHMM`) in order to use `pseudoRes`.

**Comment by AmandaPS22** [comment]
Thanks
I will try this
